src/coreclr/nativeaot/Runtime/ThunksMapping.cpp

```diff
@@ -121,8 +121,14 @@ EXTERN_C NATIVEAOT_API void* __cdecl RhAllocateThunksMapping()
         return NULL;
     }
 
-#if defined(HOST_OSX) && defined(HOST_ARM64)
+#if defined(HOST_ARM64)
+#if defined(HOST_MACCATALYST) || defined(HOST_IOS) || defined(HOST_TVOS)
+    RhFailFast(); // TODO: add pthread_jit_write_protect_np workaround
```

Comment:
I believe the workaround https://github.com/dotnet/runtime/commit/a9f1207d9f14d6bdf6e4be0f206d11df8e032382 is only possible for `MacCatalyst`

Reply:
Right, after thinking about some more we could enable the thunks support for catalyst - but I wonder if we should just use the same approach as what we'll be using for iOS/tvOS to be consistent. Then we could always RhFailFast when RhAllocateThunksMapping is called on these platforms to ensure the alternative thunks implementation is used.